code/modules/mining/mint.dm

```diff
@@ -1,198 +1,245 @@
+		if(available_num_sheets>0)
+			//available_num_sheets % sh.max_amount
+			sh.amount = available_num_sheets
+			materials.removeAmount(matID, sh.amount)
+		else
+			qdel(sh)
+	return 1
 
 /obj/machinery/mineral/mint/Topic(href, href_list)
-	if(..())
+	. = ..()
+	if(.)
 		return
 	usr.set_machine(src)
 	src.add_fingerprint(usr)
+	var/turf/out_T = get_step(src, out_dir)
+
+	if(href_list["close"])
+		usr.unset_machine(src)
+		return 1
+
 	if(processing==1)
 		to_chat(usr, "<span class='notice'>The machine is processing.</span>")
 		return
+
+	if(href_list["eject"])
+		var/datum/material/ma = materials.getMaterial(href_list["eject"])
+		var/obj/item/stack/sheet/sh = new ma.sheettype(out_T)
+		sh.amount = materials.getAmount(href_list["eject"])
+		materials.removeAmount(href_list["eject"], sh.amount)
+		if (chosen == href_list["eject"])
+			chosen = null
+
+	if("changedir" in href_list)
+		//Change_Dir()
+		var/changingdir = text2num(href_list["changedir"]) //See ore processing_unit for original comments
+		changingdir = Clamp(changingdir, 1, 2)
+
+		var/newdir = input("Select the new direction", name, "North") as null|anything in list("North", "South", "East", "West")
+		if(!newdir)
+			return 1
+		newdir = text2dir(newdir)
+
+		var/list/dirlist = list(in_dir, out_dir)
+		var/olddir = dirlist[changingdir]
+		dirlist[changingdir] = -1
+
+		var/conflictingdir = dirlist.Find(newdir)
+		if(conflictingdir)
+			dirlist[conflictingdir] = olddir
+
+		dirlist[changingdir] = newdir
+
+		in_dir = dirlist[1]
+		out_dir = dirlist[2]
+
 	if(href_list["choose"])
 		chosen = href_list["choose"]
+
 	if(href_list["chooseAmt"])
 		coinsToProduce = Clamp(coinsToProduce + text2num(href_list["chooseAmt"]), 0, 1000)
+
 	if(href_list["makeCoins"])
+		if(chosen == null)
+			return
 		var/temp_coins = coinsToProduce
-		if (src.output)
+		if (src.out_dir)
 			processing = 1
 			icon_state = "coinpress1"
-			//var/obj/item/weapon/storage/bag/money/M
 			var/datum/material/po=materials.getMaterial(chosen)
 			if(!po)
 				chosen=null
 				processing=0
 				return
 			while(materials.storage[chosen] > 0 && coinsToProduce > 0)
-			/*	if (locate(/obj/item/weapon/storage/bag/money,output.loc))
-					M = locate(/obj/item/weapon/storage/bag/money,output.loc)
-					if(M.can_be_inserted(po.cointype, 1))
-						new po.cointype(M)
-					else
-						new po.cointype(output.loc)
-				else
-				//Can't seem to be able to get the can_be_inserted check to work, would always drop the coin at the output loc
-				*/
-					//M = new/obj/item/weapon/storage/bag/money(output.loc)
-				new po.cointype(output.loc)
+				var/dest = out_T
+				if (locate(/obj/item/weapon/storage/bag/money,out_T)) //So we can put them in a bag instead of spewing them on the floor
+					dest = locate(/obj/item/weapon/storage/bag/money,out_T)
+				var/coinsleft = coins_per_sheet //coins left to produce for that single sheet.
+				while(coinsleft > 0)
+					new po.cointype(dest)
```

Comment:
Are you sure creating items inside a `/obj/item/weapon/storage`? I think there's a proc you need to call to see whether it's possible, or maybe just to check if the insertion was successful. might want to look into `/obj/item/weapon/storage/proc/can_be_inserted` and `/obj/item/weapon/storage/proc/handle_item_insertion`

Reply:
It definitely works, it's more an issue of if it's acceptable. Though I seem to have deleted the inserted check as part of the commented out moneybag code. I suppose the game was fine with it because of how the moneybag works.